src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodRetention.java

```diff
@@ -0,0 +1,15 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+public enum PodRetention {
```

Comment:
More Jenkins-y to use a `Describable/Descriptor` pattern here instead. Move implementation of `shouldDeletePod` in the Describable class so that you can implement additional behaviours through extensions.